src/d_delay.c

```diff
@@ -58,6 +59,16 @@ static void sigdelwrite_clear (t_sigdelwrite *x) /* added by Orm Finnendahl */
     memset(x->x_cspace.c_vec, 0, sizeof(t_sample)*(x->x_cspace.c_n + XTRASAMPS));
 }
 
+static void sigdelwrite_resize (t_sigdelwrite *x, t_floatarg f) /* added by Porres */
+{
+    if(f < 0)
+        f = 0;
+    if(f != x->x_deltime){
+        x->x_deltime = f;
+        sigdelwrite_clear(x);
+        sigdelwrite_updatesr(x, x->x_sr);
```

Comment:
You have to call `canvas_update_sp` at the end, so that `[delread~]` and `[delread4~]` objects have a chance to get a pointer to the new buffer! Otherwise you can easily get a segfault.